src/OmniSharp/MSBuild/ProjectFile/ProjectFileInfo.cs

```diff
@@ -175,5 +180,24 @@ public static ProjectFileInfo Create(MSBuildOptions options, ILogger logger, str
 #endif
             return projectFileInfo;
         }
+        private static LanguageVersion ToLanguageVersion(string langVersionPropertyValue)
+        {
+            if (!(string.IsNullOrWhiteSpace(langVersionPropertyValue) || langVersionPropertyValue.Equals("Default", StringComparison.OrdinalIgnoreCase)))
+            {
+                // ISO-1, ISO-2, 3, 4, 5, 6 or Default
+                switch (langVersionPropertyValue.ToLower())
+                {
+                    case "ISO-1": return LanguageVersion.CSharp1;
+                    case "ISO-2": return LanguageVersion.CSharp2;
+                    case "3": return LanguageVersion.CSharp3;
+                    case "4": return LanguageVersion.CSharp4;
+                    case "5": return LanguageVersion.CSharp5;
+                    case "6": return LanguageVersion.CSharp6;
+                }
+            }
+            return LanguageVersion.CSharp5;
```

Comment:
Updated PR. It builds and so pass tests, but I'll try to test on VSCode
proper, too.

Rafael Teixeira
O..:.)oooo

On Sun, May 3, 2015 at 3:28 AM, David Fowler <notifications@github.com>
wrote:

> In src/OmniSharp/MSBuild/ProjectFile/ProjectFileInfo.cs
> <https://github.com/OmniSharp/omnisharp-roslyn/pull/201#discussion_r29553470>
> :
>
> > +        private static LanguageVersion ToLanguageVersion(string langVersionPropertyValue)
> > +        {
> > +            if (!(string.IsNullOrWhiteSpace(langVersionPropertyValue) || langVersionPropertyValue.Equals("Default", StringComparison.OrdinalIgnoreCase)))
> > +            {
> > +                // ISO-1, ISO-2, 3, 4, 5, 6 or Default
> > +                switch (langVersionPropertyValue.ToLower())
> > +                {
> > +                    case "ISO-1": return LanguageVersion.CSharp1;
> > +                    case "ISO-2": return LanguageVersion.CSharp2;
> > +                    case "3": return LanguageVersion.CSharp3;
> > +                    case "4": return LanguageVersion.CSharp4;
> > +                    case "5": return LanguageVersion.CSharp5;
> > +                    case "6": return LanguageVersion.CSharp6;
> > +                }
> > +            }
> > +            return LanguageVersion.CSharp5;
>
> Yep Make it return LanguageVersion?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OmniSharp/omnisharp-roslyn/pull/201/files#r29553470>.
>